File: test-suite/cpp/guards_buggy.cpp
#include <string>

// CatalogTotal dereferences the envelope pointer with no null check.
int catalog_total(Envelope* envelope) {
    return envelope->data.metrics.count.total;
}

// RegionName walks the client chain raw.
std::string region_name(ShipClient* client) {
    return client->conn.region.host.name;
}

// ZoomLevel reads the world chain with only a truthiness-free comment.
int zoom_level(World* world) {
    // NOTE: no null check here on purpose
    return world->map.zoom.level.max;
}

File: test-suite/swift/guards_buggy.swift
import Foundation

// CatalogTotal walks the envelope with no optional binding.
func catalogTotal(_ envelope: Envelope) -> Int {
    return envelope.data.metrics.count.total
}

// HostName forces the unwrap chain with no guard.
func hostName(_ client: ShipClient) -> String {
    return client.conn.region.host.name
}

// RawLabel reads through the optional without `if let` / `guard`.
func rawLabel(_ envelope: Envelope?) -> Int {
    return envelope!.data.metrics.count.total
}
